.github/workflows/rust.yml

```diff
@@ -64,6 +64,12 @@ jobs:
         env:
           CARGO_HOME: "/github/home/.cargo"
           CARGO_TARGET_DIR: "/github/home/target"
+      - name: Check DataFusion Build without default features
```

Comment:
Tested in https://github.com/apache/arrow-datafusion/pull/1218 /  https://github.com/apache/arrow-datafusion/runs/4072959374?check_suite_focus=true

Reply:
You can see it fail with the same symptoms reported on the bug: https://github.com/apache/arrow-datafusion/runs/4072959374?check_suite_focus=true#step:8:99